src/Microsoft.AspNet.OData.Shared/EdmChangedObjectCollection.cs

```diff
@@ -57,5 +69,288 @@ private void Initialize(IEdmEntityType entityType)
+                            changedObjectCollection.Add(changedObject);
+                            continue;
+                        }                                              
+
+                        //Patch for addition/update. 
+                        PatchItem(deltaEntityObject, original);
+
+                        changedObjectCollection.Add(changedObj);
+                    }
+                }
+                catch(Exception ex)
+                {
+                    //Handle Failed Operation
+                    IEdmChangedObject changedObject = HandleFailedOperation(changedObj as EdmEntityObject, operation, originalObj, keys, ex.Message);
+                    
+                    Contract.Assert(changedObject != null);
+                    changedObjectCollection.Add(changedObject);
+                }
+            }
+
+            return changedObjectCollection;
+        }
+
+    
+        private static IDictionary<string, object> GetKeyValues(List<IEdmStructuralProperty> keys, IEdmChangedObject changedObj)
+        {
+            IDictionary<string, object> keyValues = new Dictionary<string, object>();
+
+            foreach (IEdmStructuralProperty key in keys)
+            {
+                object value;
+                changedObj.TryGetPropertyValue(key.Name, out value);
+
+                if (value != null)
+                {
+                    keyValues.Add(key.Name, value);
+                }
+            }
+
+            return keyValues;
+        }
+
+        private void PatchItem(EdmStructuredObject changedObj, EdmStructuredObject originalObj)
+        {
+            foreach (string propertyName in changedObj.GetChangedPropertyNames())
+            {
+                ApplyProperties(changedObj, originalObj, propertyName);
+            }
+
+            foreach (string propertyName in changedObj.GetUnchangedPropertyNames())
+            {
+                ApplyProperties(changedObj, originalObj, propertyName);
+            }
+        }
+
+        private void ApplyProperties(EdmStructuredObject changedObj, EdmStructuredObject originalObj, string propertyName)
+        {
+            object value;
+            if (changedObj.TryGetPropertyValue(propertyName, out value))
+            {
+                EdmChangedObjectCollection changedColl = value as EdmChangedObjectCollection;
+                if (changedColl != null)
+                {
+                    TypelessPatchMethodHandler patchHandler = PatchHandler.GetNestedPatchHandler(originalObj, propertyName);
+                    if (patchHandler != null)
+                    {
+                        changedColl.Patch(patchHandler);
+                    }
+                    else
+                    {
+                        object obj;
+                        originalObj.TryGetPropertyValue(propertyName, out obj);
+
+                        ICollection<EdmStructuredObject> edmColl = obj as ICollection<EdmStructuredObject>;
```

Comment:
Will the call to `originalObj.TryGetPropertyValue` above always result into a non-null value for `obj`? If no by any chance, then we'll end up with a null in the collection

Reply:
It shuld create a new collection then,